### PR TITLE
Do not show race interstitial if geolocation fails

### PR DESCRIPTION
**Pull request by ewjordan**
This PR changes some of the logic around showing the race interstitial to tighten it up, and show it to as few people outside of the country as possible:

- Switch to using Devise-determined IP instead of request.remote_ip (which seems less reliable) (EDIT: turns out the actual bug was elsewhere, but involved forgetting to pass the IP through the controller, and just using the user field avoids that issue)
- Default to not showing dialog if IP is missing or geolocation fails
- Update tests

### CODE REVIEW

**Comment by ashercodeorg** [comment]
Generally, I'm against production hacks to facilitate testing. But if we are going to allow forcing a response via `allowed`, please document this clearly in method comments.

Beyond that, LGTM.

File: dashboard/app/models/user.rb [hunk 1]
```diff
@@ -1083,6 +1083,7 @@ def should_see_inline_answer?(script_level)
   end
 
   def show_race_interstitial?(ip = nil)
-    RaceInterstitialHelper.show_race_interstitial?(self, ip)
+    ip_to_check = ip || @current_sign_in_ip
```
Comment (ashercodeorg) [inline dashboard/app/models/user.rb]: Why the `@` in `@current_sign_in_ip`?

**Comment by wjordan** [comment]
Parts 2 and 3 LGTM, however I'm not yet convinced about part 1 ('Switch to using Devise-determined IP instead of request.remote_ip (which seems less reliable)'). Under the hood, Devise uses [`request.remote_ip`](https://github.com/plataformatec/devise/blob/master/lib/devise/models/trackable.rb#L23) to determine the user's current IP address, so I'm suspicious that Devise-determined IP is at all more reliable. I would like some more investigation/testing on this point before committing code changes based on the assumption.

File: dashboard/app/views/scripts/show.html.haml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 -# This variable will get any data we need to pass along to scriptOverview.js
 - scriptOverviewData = { scriptData: @script.summarize }
 
-- if current_user && current_user.show_race_interstitial?
```
Comment (ewjordan) [inline dashboard/app/views/scripts/show.html.haml:4]: This was the real bug here, I missed updating this when I updated the index.html.haml version

**Pull request merged**